firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java

```diff
@@ -780,18 +811,21 @@ boolean finalizeSessions(int maxCustomExceptionEvents) {
    * class.
    */
   private void doOpenSession() throws Exception {
-    final Date startedAt = new Date();
+    final long startedAtSeconds = new Date().getTime() / 1000;
     final String sessionIdentifier = new CLSUUID(idManager).toString();
 
     Logger.getLogger().d(Logger.TAG, "Opening a new session with ID " + sessionIdentifier);
 
     nativeComponent.openSession(sessionIdentifier);
 
-    writeBeginSession(sessionIdentifier, startedAt);
+    writeBeginSession(sessionIdentifier, startedAtSeconds);
     writeSessionApp(sessionIdentifier);
     writeSessionOS(sessionIdentifier);
     writeSessionDevice(sessionIdentifier);
     logFileManager.setCurrentSession(sessionIdentifier);
+
+    // Firebase Crashlytics requires session IDs without dashes.
+    reportingCoordinator.onBeginSession(sessionIdentifier.replaceAll("-", ""), startedAtSeconds);
```

Comment:
The fact that this needed a comment makes me wonder: Is this worth pulling into a static method? A clever name could reduce the need for a javadoc

Reply:
Yeah, that's fair.